### DISCUSSION TITLE
Simulating growth error

### DISCUSSION THREAD

**Discussion by clauSilvaA** [micom-dev/micom#133] [Q&A] [answered]
Dear:
It's my first time using micom. I Built my community model. when I try to Simulating growth I have the next error:

ValueError: No ID from the medium could be found in the exchange reactions. This means you probably have mismatched IDs...

The command that I used:

from micom.workflows import build
man=build(data2, out_folder="models", model_db=db, cutoff=0.0001, threads=2)

from micom.qiime_formats import load_qiime_medium
medium = load_qiime_medium(test_medium)

from micom.workflows import grow
res = grow(man, model_folder="/home/lalo/Desktop/MICOM/", medium=medium, tradeoff=0.5, threads=6)

and I tried this medium too:

medium = load_qiime_medium("western_diet_gut.qza")

obtaining the same result. 
How I can solve this error?
I used a few bacterial strains from Agora in a new database created by my

**Comment by cdiener** [comment]
Hi, thanks for trying out MICOM! Sorry you encountered an error. For `grow` to work the metabolite names in your community model and medium have to match. If you used any of the prebuilt model databases that means you usually need a medium that is compatible. I am currently changing the medium repository to make this more obvious, so apologies for the confusion here. Most of the media in the [media repository](https://github.com/micom-dev/media) work with AGORA only. What model database did you use? Maybe I can point you towards the correct medium.

**Reply by clauSilvaA** [reply]
I use some examples from AGORA database

**Reply by cdiener** [reply]
Hmm that is odd. You could try again with the version from [here](https://github.com/micom-dev/media/raw/main/media/western_diet_gut_agora.qza), but it is unlikely to work if the previous one did not. How was the database created? Was some renaming done on the metabolites or something?

**Reply by clauSilvaA** [reply]
I created the database with this command:

manifest=build_database(db, out_path="/home/lalo/Desktop/MICOM/newdb.zip", rank='species', threads=6,  progress=True)

my

**Reply by clauSilvaA** [reply, marked as answer]
Dear, I found my error. It was a corrupt xml file. Thanks you for your support

**Reply by cdiener** [reply]
Thanks for sending the model database as well that allowed to check it out a bit more. I also noticed that you have two Lactobacillus species in there that were not part of the AGORA reconstructions. Those won't play well with the other models since they don't use the same IDs for metabolites and those also have missing annotations for the transport reactions. The species (though not the exact strains) are part of AGORA 2, so maybe you could try with the species models from there (https://zenodo.org/record/7739096, for instance agora201_refseq216_species_1.qza).
